pkg/types/dsse/dsse.go

```diff
@@ -102,46 +116,50 @@ func verifyEnvelope(verifiers map[*pb.Verifier]verifier.Verifier, pbenv *pbdsse.
 		allSigs[sig.Sig] = struct{}{}
 	}
 
+	dsseHashAlgSet := make(map[crypto.Hash]v1.HashAlgorithm)
+
 	for v, verifierKey := range verifiers {
 		if len(allSigs) == 0 {
 			break // if all signatures have been verified, do not attempt anymore
 		}
 
 		algDetails, err := signature.GetAlgorithmDetails(v.KeyDetails)
 		if err != nil {
-			return nil, fmt.Errorf("getting key algorithm details: %w", err)
+			return nil, nil, fmt.Errorf("getting key algorithm details: %w", err)
 		}
 		alg := algDetails.GetHashType()
+		dsseHashAlgSet[alg] = algDetails.GetProtoHashType()
```

Comment:
oh yeah, I dunno if this is ideal, or we should've created the algorithm mapping separately. This feels a bit goofy, but I can live with it.

Reply:
Yea, given it's for this one special case, I'm fine with this. Multi-sigs types aren't the norm (if we were to add more).